### ISSUE TITLE
resnet on imagenet

### ISSUE THREAD

**Issue by a-maci** [ppwwyyxx/tensorpack#38]
Not an issue per se but more of a question: could you provide resent configs (34, 50, 101, 1001, etc) for training on imagenet dataset?

Thanks

**Comment by ppwwyyxx** [comment]
Code is updated, you should get the following top1 single-crop error with the config: 
ResNet18: 30.5%; ResNet34: 27.0%; ResNet50: 24.5%.
It's a little bit worse than fb.resnet.torch, so I'm still investigating.

**Comment by a-maci** [comment]
Thanks a lot. I will try out your code.
Closing this issue for now.

**Comment by ppwwyyxx** [comment]
The code earlier has some bug on performance. The updated code now should produce the numbers I mentioned above (or better).

**Comment by a-maci** [comment]
Thanks

**Comment by a-maci** [comment]
How many GPU cores did you use for training each of these ResNet configurations? 
I am not able to get to the same accuracy level that you mention for ResNet-34 with 2 GPUs.

In general, I have found the accuracy (which depends on learning rate schedule) to be quite sensitive to the number of GPUs. Is this your observation as well? Any workaround/tips/tricks you use to get around this issue?

Thanks

**Comment by ppwwyyxx** [comment]
The total batch size is fixed so it shouldn't be very sensitive to the number of GPUs.
Some of the models are trained with 2 GPUs and some with 4. I couldn't remember which is which.
What's the number you get?

**Comment by a-maci** [comment]
I see. 
I had changed the batch size to fit on 2 GPUs. I am getting 0.65% higher error rate than you reported.

**Comment by ppwwyyxx** [comment]
On imagenet larger batch usually help improve the model. If you made it smaller to fit on your GPU it may get worse.
I'm only using the batch size used by fb.resnet.torch, it probably will train better with a larger batch size.

There're some other discussions [here](https://github.com/tornadomeet/ResNet/issues/28) about batch size. They tried larger batch size.
